atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableRendererV2.java

```diff
@@ -0,0 +1,422 @@
+            results.add(renderGetTableRef());
+            results.add(renderGetTableName());
+            results.add(renderGetNamespace());
+            if (!tableMetadata.getColumns().hasDynamicColumns()) {
+                results.addAll(renderNamedGet());
+                results.add(renderNamedDeleteRow());
+                results.addAll(renderNamedPut());
+            }
+            return results;
+        }
+
+        private MethodSpec renderGetRawTableName() {
+            return MethodSpec.methodBuilder("getRawTableName")
+                    .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                    .addStatement("return rawTableName")
+                    .returns(String.class)
+                    .build();
+        }
+
+        private MethodSpec renderGetTableName() {
+            return MethodSpec.methodBuilder("getTableName")
+                    .addModifiers(Modifier.PUBLIC)
+                    .addStatement("return tableRef.getQualifiedName()")
+                    .returns(String.class)
+                    .build();
+        }
+
+        private MethodSpec renderGetTableRef() {
+            return MethodSpec.methodBuilder("getTableRef")
+                    .addModifiers(Modifier.PUBLIC)
+                    .addStatement("return tableRef")
+                    .returns(TableReference.class)
+                    .build();
+        }
+
+        private MethodSpec renderGetNamespace() {
+            return MethodSpec.methodBuilder("getNamespace")
+                    .addModifiers(Modifier.PUBLIC)
+                    .addStatement("return tableRef.getNamespace()")
+                    .returns(Namespace.class)
+                    .build();
+        }
+
+        private List<MethodSpec> renderNamedGet() {
+            ArrayList<MethodSpec> getterResults = new ArrayList<>();
+            for (NamedColumnDescription col : ColumnRenderers.namedColumns(tableMetadata)) {
+                getterResults.add(renderNamedGetColumn(col));
+                if (tableMetadata.getRowMetadata().getRowParts().size() == 1) {
+                    getterResults.add(renderNamedGetSeveralRowsColumn(col));
+                }
+                getterResults.add(renderNamedGetAllRowsColumn(col));
+            }
+
+            return getterResults;
+        }
+
+        private MethodSpec renderNamedGetColumn(NamedColumnDescription col) {
+            MethodSpec.Builder getterBuilder = MethodSpec.methodBuilder("get" + VarName(col))
+                    .addModifiers(Modifier.PUBLIC);
+
+            getterBuilder = addParametersFromRowComponents(getterBuilder, tableMetadata);
+
+            getterBuilder.returns(ParameterizedTypeName.get(
+                    ClassName.get(Optional.class),
+                    ClassName.get(col.getValue().getValueType().getTypeClass())));
+            getterBuilder
+                    .addStatement("$T row = $T.of($N)", rowType, rowType, getArgumentsFromRowComponents(tableMetadata))
```

Comment:
should this be `$L` instead of `$N`?

Reply:
(there also are a few other usages of `$N` where I have the same question)

Reply:
Thanks for the catch!